File: vaccination-frontend/src/app/login/model/user-dto.ts
export class UserDTO {
    private _dni!: string;
    private _email!: string;
    private _name!: string;
    private _roles!: string[];
    private _surname!: string;

    public get dni(): string {
        return this._dni;
    }
    public get email(): string {
        return this._email;
    }
    public get name(): string {
        return this._name;
    }
    public get roles(): string[] {
        return this._roles;
    }
    public get surname(): string {
        return this._surname;
    }

    public set dni(_dni: string) {
        this._dni = _dni;
    }
    public set email(_email: string) {
        this._email = _email;
    }
    public set name(_name: string) {
        this._name = _name;
    }
    public set roles(_roles: string[]) {
        this._roles = _roles;
    }
    public set surname(_surname: string) {
        this._surname = _surname;
    }
}

File: vaccination-backend/src/main/java/com/kruger/vaccination/DTO/UserDTO.java
package com.kruger.vaccination.DTO;

import java.util.Set;

import javax.validation.constraints.Email;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.Size;

import lombok.Data;

@Data
public class UserDTO {

    @NotBlank(message = "No se ingresó una cédula")
    private String dni;

    @Email
    @NotBlank(message = "No se ingresó un email")
    private String email;

    private Long id;

    @NotBlank(message = "No se ingresó el nombre o nombres")
    @Size(message = "El nombre debe tener al menos 3 caracteres", min = 3)
    private String name;

    private Set<String> roles;

    @NotBlank(message = "No se ingresó el apellido o apellidos")
    @Size(message = "El apellido debe tener al menos 3 caracteres", min = 3)
    private String surname;

    private String username;

    public UserDTO() {
    }

    public UserDTO(String dni, String email, Long id, String name, Set<String> roles, String surname, String username) {
        this.dni = dni;
        this.email = email;
        this.id = id;
        this.name = name;
        this.roles = roles;
        this.surname = surname;
        this.username = username;
    }
}